lldb/test/Shell/SymbolFile/DWARF/delayed-definition-die-searching.test

```diff
@@ -0,0 +1,40 @@
+# Test definition DIE searching is delayed until complete type is required.
+
+# RUN: split-file %s %t
+# RUN: %clangxx_host %t/main.cpp %t/t1_def.cpp -g -gsimple-template-names -o %t.out
+# RUN: %lldb -b %t.out -s %t/lldb.cmd | FileCheck %s
+
+# CHECK:      (lldb) p v1
+# CHECK-NEXT: DWARFASTParserClang::ParseTypeFromDWARF{{.*}}DW_TAG_structure_type (DW_TAG_structure_type) name = 't2'
```

Comment:
CHECK-NEXT on log output sounds like a very bad idea, as the test can be broken by adding a random log statement. For what you're trying to do, a plain CHECK should suffice (though i'd also consider if this can be checked using something less volatile than log messages -- for example, you might be able to use `image dump ast` to compare the clang ast of the type before/after completion).

Reply:
Updated to use `CHECK`. Unfortunately, it doesn't work with `image dump ast` because this change doesn't affect  when type completion happens. So, after `p v1`, w/wo this change, lldb just creates `ClassTemplateSpecializationDecl` for `t1<int>` with no definition. When doing `p v2', lldb tries to complete the `t1<int>` decl. This change only moves the definition searching to the time when completing a type.

Reply:
I see, that makes sense. Thanks for explaining.